Functions/QualityControl/xASL_qc_ParsePdfConfig.m

```diff
@@ -0,0 +1,675 @@
+        return
+    end 
+
+    if ~isfield(x.Output.(qcStruct.module),( qcStruct.parameter)) 
+        fprintf (['QC Value ' x.Output.(qcStruct.module).(qcStruct.parameter) ' not found, skipping. \n']);
+        return
+    end
+
+    TempValue = x.Output.(qcStruct.module).(qcStruct.parameter);
+    
+    % Translate from the provided translation tsv if that's enabled
+    if settingsPDF.QC_TSV_Translations
+        qcStruct = xASL_qc_ParsePdfConfig_sub_QC_Translation(qcStruct, settingsPDF);
+    end
+
+    % Check if the QC has an alias, if so replaced the parameter with the alias in the printed text.
+    if ~isfield(qcStruct, 'alias') || isempty(qcStruct.alias)
+        qcStruct.alias = qcStruct.parameter;
+    end
+
+    % Check if the QC has a unit, if not set it to an empty string.
+    if ~isfield(qcStruct, 'unit') 
+        qcStruct.unit = '';
+    end
+
+    % Check if the QC has a range, if so check if the value is within the range, else print in red.
+    if isfield(qcStruct, 'range') 
+        if ~isempty(qcStruct.range)
+            [range] = strsplit(qcStruct.range, '-');
+            if (TempValue < xASL_str2num(range{1})) || (TempValue > xASL_str2num(range{2}))
+                settingsPDF.color = 'r';
+            end
+            qcStruct.range = ['(' qcStruct.range ')'];
+        end
+    else
+        qcStruct.range = '';
+    end
+
+    % Convert the value to a string.
+    TempValue = xASL_num2str(TempValue);
+
+
+    qcStruct.alias = xASL_qc_ParsePdfConfig_sub_PaddedString( qcStruct.alias, settingsPDF.QC_string_limit);
+    TempValue = xASL_qc_ParsePdfConfig_sub_PaddedString( TempValue, 12, 'right');
+    UnitRange = xASL_qc_ParsePdfConfig_sub_PaddedString( [qcStruct.unit, qcStruct.range], 12);
+
+    % Combine name, value, unit and range into a single string for printing.
+    if size(TempValue, 1) == 1
+        string = sprintf([qcStruct.alias ':', TempValue, ' ', UnitRange, ' \n']);
+    end
+
+end
+
+function [struct] = xASL_qc_ParsePdfConfig_sub_QC_Translation(struct, settingsPDF)
+    % This function replaces QC values with long names with easier to read names with units.
+    name = struct.parameter;
+    index = find(strcmp(settingsPDF.QC_Translation(:,1), name));
+    if ~isempty(index)
+        struct.alias = char(settingsPDF.QC_Translation(index, 2));
+        struct.unit  = char(settingsPDF.QC_Translation(index, 3));
+        struct.range = char(settingsPDF.QC_Translation(index, 4));
+    end
+end
+
+function resultText = xASL_qc_ParsePdfConfig_sub_PaddedString(textToPrint, textWidth, align, SymbolToFill)
+    if nargin < 3 || isempty(align)
+        align = 'left';
+    end
+
+    if nargin < 4 || isempty(SymbolToFill)
+        SymbolToFill = ' ';
+    end
+
+    %% Create default string
+    resultText = repmat(SymbolToFill, 1, textWidth);
+    
+    % Pad spaces
+    if ~isempty(textToPrint)
+        textToPrint = [textToPrint ' '];
```

Comment:
Here we sometimes have char arrays that are not only horizontal, because a parameter can have multiple values apparently, and then the values should be concatenated in a string/char or just the first reported (depending on the length).

E.g., I had here:

textToPrint =

  2×4 char array

    '1710'
    '3142'

First, I would never use char arrays, can we get rid of them?

Because if we have 

textToPrint2 = {'1710';'3142'}

textToPrint2 =

  2×1 cell array

    {'1710'}
    {'3142'}

this is easy to concatenate iterating over cells (2 elements)
With a char array, this example would be 8 elements.

Char arrays is something new of Matlab I think, at least Jan and I never used them.

Or if it becomes very tricky to get rid of char arrays, perhaps you can think of a different solution?

Reply:
I have decided to leave it as Char as it is, I've reworked the way it reads values, and it now horizontally concatenates them. Sadly for chars a simple var= var(:)' does not work due to indexing of chars, so I used append instead. This can become slow if a **very large** amount of values need to be concatenated.  This can be relieved by pre-defining the character array length and replacing values inside a predefined array, but realistically it won't ever matter for this purpose.